runtime/src/main/java/org/corfudb/util/NetworkUtils.java

```diff
@@ -27,13 +51,16 @@
      */
     public static String getAddressFromInterfaceName(String interfaceName) {
         try {
-            Enumeration<InetAddress> inetAddress = NetworkInterface.getByName(interfaceName)
-                    .getInetAddresses();
-            InetAddress currentAddress;
-            while (inetAddress.hasMoreElements()) {
-                currentAddress = inetAddress.nextElement();
-                if (currentAddress instanceof Inet4Address) {
-                    return currentAddress.getHostAddress();
+            for (NetworkInterface networkInterface : getAllInterfaces()) {
+                if (networkInterface.isUp()
```

Comment:
This probably needs to be reflected in the comment.

Reply:
done.